parlai/core/worlds.py

```diff
@@ -631,6 +631,8 @@ def episode_done(self):
         return False
 
     def epoch_done(self):
+        if not self.random:
+            return any([world.epoch_done() for world in self.worlds])
```

Comment:
I'm not sure about this.

Before: "return done if all worlds are done"
now: "if we aren't doing random sampling, return done if any world is done"

however, there was an implicit "if we aren't doing random sampling" already, since random sampling never returns epoch_done. so the logic here has been flipped from "all" to "any". I'm worried this will prevent all examples from being covered.